app/src/main/java/de/danoeh/antennapod/dialog/ShareDialog.java

```diff
@@ -0,0 +1,121 @@
+package de.danoeh.antennapod.dialog;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.preference.PreferenceManager;
+import android.view.View;
+import android.widget.RadioButton;
+import android.widget.RadioGroup;
+import android.widget.Switch;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+import androidx.fragment.app.DialogFragment;
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.core.feed.FeedItem;
+import de.danoeh.antennapod.core.preferences.UserPreferences;
+import de.danoeh.antennapod.core.util.ShareUtils;
+
+public class ShareDialog extends DialogFragment {
+
+    private static final String TAG = "ShareDialog";
+    private final Context ctx;
+    private FeedItem item;
+
+    private RadioGroup radioGroup;
+    private RadioButton radioEpisodeWebsite;
+    private RadioButton radioMediaFile;
+    private Switch switchStartAt;
+    private SharedPreferences prefs;
+
+    public ShareDialog(Context ctx, FeedItem item) {
+        this.ctx = ctx;
+        this.item = item;
+        prefs = PreferenceManager.getDefaultSharedPreferences(ctx);
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+        View content = View.inflate(ctx, R.layout.share_episode_dialog, null);
+        AlertDialog.Builder builder = new AlertDialog.Builder(ctx);
+        builder.setTitle(R.string.share_episode_label);
+        builder.setView(content);
+
+        radioGroup = content.findViewById(R.id.share_dialog_radio_group);
+        radioEpisodeWebsite = content.findViewById(R.id.share_episode_website_radio);
+        radioMediaFile = content.findViewById(R.id.share_media_file_radio);
+        switchStartAt = content.findViewById(R.id.share_start_at_timer_dialog);
+
+        setupOptions();
+
+        builder
+                .setPositiveButton(R.string.share_episode_positive_label_button, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int id) {
+                        if (radioEpisodeWebsite.isChecked()) {
+                            if (switchStartAt.isChecked()) {
+                                ShareUtils.shareFeedItemLink(ctx, item, true);
+                                prefs.edit().putBoolean(UserPreferences.PREF_SHARE_EPISODE_START_AT, true).apply();
```

Comment:
Please decouple this from the UserPreferences class (that class is already huge). Just create the constants here -- they are not needed globally.

Reply:
perfect, I have already update this for the next commit. Thanks